vector/src/main/java/im/vector/riotx/features/attachments/AttachmentsHelper.kt

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2019 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package im.vector.riotx.features.attachments
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import com.kbeanie.multipicker.api.Picker.*
+import com.kbeanie.multipicker.core.PickerManager
+import com.kbeanie.multipicker.utils.IntentUtils
+import im.vector.matrix.android.api.session.content.ContentAttachmentData
+import im.vector.riotx.core.platform.Restorable
+import timber.log.Timber
+
+private const val CAPTURE_PATH_KEY = "CAPTURE_PATH_KEY"
+private const val PENDING_TYPE_KEY = "PENDING_TYPE_KEY"
+
+/**
+ * This class helps to handle attachments by providing simple methods.
+ * The process is asynchronous and you must implement [Callback] methods to get the data or a failure.
+ */
+class AttachmentsHelper private constructor(private val pickerManagerFactory: PickerManagerFactory) : Restorable {
+
+    companion object {
+        fun create(fragment: Fragment, callback: Callback): AttachmentsHelper {
+            return AttachmentsHelper(FragmentPickerManagerFactory(fragment, callback))
+        }
+
+        fun create(activity: Activity, callback: Callback): AttachmentsHelper {
+            return AttachmentsHelper(ActivityPickerManagerFactory(activity, callback))
+        }
+    }
+
+    interface Callback {
+        fun onContactAttachmentReady(contactAttachment: ContactAttachment) {
+            Timber.v("On contact attachment ready: $contactAttachment")
```

Comment:
Privacy: please do not log user data or protect with the correct if (LOG_PRIVACY...)